// components/MakeTable.js
import React, {Component} from 'react';
import actionTable from 'actions/actionTable';
import {KeyNames} from 'constants/NewEvent';
import './App.css';

class MakeTd extends Component{
    details(){
        console.log('click td');
    }
    render(){
        return(
            <td key = {this.props.index} onClick={() => this.details()}>
                <Link className='TdLink' to ={`/detail/${this.props.index}`} >
                    {
                        this.props.name
                    }
                </Link>
            </td>
        );
    }
}   

class MakeTr extends Component{
    constructor(props){
        super(props);
        props = this.props.columns;

    }

    createNewArr(){
        let arr = [];
        for(let key in this.props.columns){
            if(KeyNames.has(key)){
                if(key === 'rating'){
                    arr.push(this.props.columns['rating']['average'] || '0');
                } else {
                    arr.push(this.props.columns[key]);
                }
            }
        }
        return arr;
    }

    newValueTr(){
        return this.createNewArr().map((el)=> <MakeTd name = {el} index = {this.props.index} />);
    }
    


    render()
    
    {
        return(
            <tr>
                { this.newValueTr() }
            </tr>
            
        );
    }
} 

class MakeTable extends Component{
    componentWillMount(){
        this.props.actionTable();
        // console.log(this.props);

    }
    chooseCol(){
        if (this.props.page !== null && this.props.page !== false) {
            return this.props.page.map((row, i)=> <MakeTr key={i} columns = {row.show } index ={i} />);
        }
    }

    createTh(){
        return Array.from(KeyNames).map((value, i, arr)=> <th onClick={() =>{
            if(value === 'name'){
                console.log('name');
                console.log(arr);
                this.props.page.sort(function (a, b) {
                    if (a.show.name > b.show.name) {
                        return 1;
                    }
                    if (a.show.name < b.show.name) {
                        return -1;
                    }
                    else {
                        return 0;
                    }
                });
            } 
            this.props.action(this.props.page); // запись отсортированного массива в стор(осталось перезапустить table с отсортированным массивом)
            if(value === 'rating'){
                console.log('rating');
            }
        }} key = {i}>{value.toUpperCase()}</th>);
    }

    render()
    {
        return(
            <table className='infoBlock'>
                <tbody>
                    <tr>
                        {
                            this.createTh() 
                        }
                    </tr>
                    {
                      
                        this.chooseCol()
                    }
                </tbody>
            </table>
        );
    }
}

export default MakeTable;
