// pages/Login.js
import {Component} from 'react';
import {Link} from 'react-router-dom';
import store from 'store';
import { GraphQLClient } from 'graphql-request';
const gql = new GraphQLClient('http://localhost:4000/graphql', { headers: {} });

class Login extends Component {

    loginUser(){

        gql.request(
            `query getUser ($login: String!, $password: String!) {
                user (login: $login, password: $password) {
                login
                password
                }
                
            }`,
            {  
                login: this.login.value,
                password: this.password.value}
        )
            .then(data => {
                console.log(data.user);
                return store.dispatch({type: 'ADD_USER', data: {user:data.getUser}});
            });
        // console.log();
        this.login.value = '';
        this.password.value = '';
    }
    render() {
        return (
   
            <div className=" App-header App SectionApp">
                <h1>Login please!</h1><br /><br />
                <input className="InputSpace" placeholder='Type your login'  type="text" ref={(input) => { this.login = input; }} /><br /><br />
                <input className="InputSpace" placeholder='Type your password' type="password" name="password"  ref={(input) => { this.password = input; }} /><br /><br />
                <button onClick={this.loginUser.bind(this)}>Login user</button><br /><br />
                <Link to='/' className='WhiteLink'>Home</Link>&nbsp;&nbsp;
                <Link to='/registration' className='WhiteLink'>Registration</Link>
            </div>
    
        );
    }
}

export default Login;
