// pages/Registration.js
import {Component} from 'react';
import {Link} from 'react-router-dom';
import store from 'store';
import { GraphQLClient } from 'graphql-request';
const gql = new GraphQLClient('http://localhost:4000/graphql', { headers: {} });

class Registration extends Component {

    addUser() {

        gql.request(
            `mutation createUser($login: String!, $mail:String!, $password: String!, 
                $firstName: String!, $lastName: String!, 
                $dateOfBirth: String!, $phoneNumber: String!) {
                createUser(login: $login, mail: $mail, password: $password, 
                    firstName: $firstName, lastName: $lastName, 
                    dateOfBirth: $dateOfBirth, phoneNumber: $phoneNumber) {
                        login
                        mail
                        password
                         firstName
                         lastName
                         phoneNumber
                         dateOfBirth
                }
            }`,
            {login: this.login.value,
                mail: this.mail.value,
                password: this.password.value,
                firstName: this.firstName.value,
                lastName: this.lastName.value,
                dateOfBirth: this.dateOfBirth.value,
                phoneNumber: this.phoneNumber.value}
        )
            .then(data => store.dispatch({type: 'ADD_USER', data: {user:data.createUser}}));

        this.login.value = '';
        this.mail.value = '';
        this.password.value = '';
        this.firstName.value = '';
        this.lastName.value = '';
        this.dateOfBirth.value = '';
        this.phoneNumber.value = '';
    }

            

    render() {
        return (

            <div className=" App-header App SectionApp">
                <h1>Create User</h1><br /><br />
                <input className="InputSpace" placeholder='Type your login'  type="text" ref={(input) => { this.login = input }} /><br /><br />
                <input className="InputSpace" placeholder='Type your mail'  type="text" ref={(input) => { this.mail = input }} /><br /><br />
                <input className="InputSpace" placeholder='Type your firstName'  type="text" ref={(input) => { this.firstName = input }} /><br /><br />
                <input className="InputSpace" placeholder='Type your lastName'  type="text" ref={(input) => { this.lastName = input }} /><br /><br />
                <input className="InputSpace" placeholder='Type your password'  type="text" ref={(input) => { this.password = input }} /><br /><br />
                <input className="InputSpace" placeholder='Type your dateOfBirth'  type="text" ref={(input) => { this.dateOfBirth = input }} /><br /><br />
                <input className="InputSpace" placeholder='Type your phoneNumber'  type="text" ref={(input) => { this.phoneNumber = input }} /><br /><br />
                <button onClick={this.addUser.bind(this)}>Add user</button><br /><br />
                <Link to='/' className='WhiteLink'>Home</Link>&nbsp;&nbsp;
            </div>
            
        );
    }
}

export default Registration;
